nixos/modules/services/misc/pinchflat.nix

```diff
@@ -0,0 +1,162 @@
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 8945;
+        description = "Port on which the Pinchflat web interface is available.";
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Open ports in the firewall for the Pinchflat web interface";
+      };
+
+      selfhosted = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Use a weak secret. If true, you are not required to provide a {env}`SECRET_KEY_BASE` through the `secretsFile` option. Do not use this option in production!";
+      };
+
+      logLevel = mkOption {
+        type = types.enum [
+          "debug"
+          "info"
+          "warning"
+          "error"
+        ];
+        default = "info";
+        description = "Log level for Pinchflat.";
+      };
+
+      extraConfig = mkOption {
+        type =
+          with types;
+          attrsOf (
+            nullOr (oneOf [
+              bool
+              int
+              str
+            ])
+          );
+        default = { };
+        example = literalExpression ''
+          {
+            YT_DLP_WORKER_CONCURRENCY = 1;
+          }
+        '';
+        description = ''
+          The configuration of Pinchflat is handled through environment variables.
+          The available configuration options can be found in [the Pinchflat README](https://github.com/kieraneglin/pinchflat/README.md#environment-variables).
+        '';
+      };
+
+      secretsFile = mkOption {
+        type = with types; nullOr path;
+        default = null;
+        example = "/run/secrets/pinchflat";
+        description = ''
+          Secrets like {env}`SECRET_KEY_BASE` and {env}`BASIC_AUTH_PASSWORD`
+          should be passed to the service without adding them to the world-readable Nix store.
+
+          Note that either this file needs to be available on the host on which `pinchflat` is running,
+          or the option `selfhosted` must be `true`.
+          Further, {env}`SECRET_KEY_BASE` has a minimum length requirement of 64 bytes.
+          One way to generate such a secret is to use `openssl rand -hex 64`.
+
+          As an example, the contents of the file might look like this:
+          ```
+          SECRET_KEY_BASE=...copy-paste a secret token here...
+          BASIC_AUTH_USERNAME=...basic auth username...
+          BASIC_AUTH_PASSWORD=...basic auth password...
+          ```
+        '';
+      };
+
+      package = mkPackageOption pkgs "pinchflat" { };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.selfhosted || !builtins.isNull cfg.secretsFile;
+        message = "Either `selfhosted` must be true, or a `secretsFile` must be configured.";
+      }
+    ];
+
+    systemd.services.pinchflat = {
+      description = "pinchflat";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      path = [
```

Comment:
I don't really see why yt-dlp / apprise are both here and in the package runtimeDeps. Package runtimeDeps should be enough?

Reply:
It's definitely required in the service paths, but can be ommitted in the runtimeDeps - though in that case, someone installing the package and not using the module would also require the other two. Not sure how likely that is; if you want me to remove the runtimeDeps, I have no problem with that, though personally I would just leave them in.

Reply:
Yes, this is the part I don't understand.
If the package is usable *without* installing additional packages, you wouldn't need to add the packages here.

If I remember correctly, runtimeDeps is more about libraries, and not applications. To make applications available on $PATH you would usually use makeWrapper / wrapProgram.

Reply:
Ahhh, makes sense. `wrapProgram` seems to have been the right call, nice.